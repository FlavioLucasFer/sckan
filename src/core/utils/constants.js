const MAIN_COLOR = '#1ac3d6';
const SECUNDARY_COLOR = '#245257';

export {
    MAIN_COLOR,
    SECUNDARY_COLOR,
};