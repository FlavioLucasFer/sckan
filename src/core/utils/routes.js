const HOME_ROUTE = '/';
const LOGIN_ROUTE = '/login';
const COMPANY_ROUTE = '/companies';
const COMPANY_FORM_ROUTE = `${COMPANY_ROUTE}/form`;
const PROJECT_ROUTE = '/projects';
const PROJECT_FORM_ROUTE = `${PROJECT_ROUTE}/form`;
const SPRINT_ROUTE = '/sprints';
const SPRINT_FORM_ROUTE = `${SPRINT_ROUTE}/form`;
const TASK_ROUTE = '/tasks';
const TASK_FORM_ROUTE = `${TASK_ROUTE}/form`;
const USER_ROUTE = '/users';
const USER_FORM_ROUTE = `${USER_ROUTE}/form`;
const USER_PROFILE = '/user';
const USER_SITTINGS = `${USER_PROFILE}/settings`;

export {
	HOME_ROUTE,
  LOGIN_ROUTE,
  COMPANY_ROUTE,
	COMPANY_FORM_ROUTE,
  PROJECT_ROUTE,
	PROJECT_FORM_ROUTE,
  SPRINT_ROUTE,
	SPRINT_FORM_ROUTE,
  TASK_ROUTE,
	TASK_FORM_ROUTE,
  USER_ROUTE,
	USER_FORM_ROUTE,
	USER_PROFILE,
	USER_SITTINGS,
};