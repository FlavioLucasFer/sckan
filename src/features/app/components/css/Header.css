.nav-wrapper {
	line-height: 50px !important;
}

.nav-ul>li>i {
	cursor: pointer;
	line-height: 50px !important;
	margin: 0px 10px;
}

.brand-logo {
	font-weight: bold;
	cursor: pointer;
}