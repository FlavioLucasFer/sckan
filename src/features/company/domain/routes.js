'use strict';

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/companies`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	REMOVE: `${BASE_URL}/remove`,
};

export default ROUTES;
