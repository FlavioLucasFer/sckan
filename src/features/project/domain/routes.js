'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/projects`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
};

export default ROUTES;
