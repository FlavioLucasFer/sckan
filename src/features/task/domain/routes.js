'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/tasks`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
};

export default ROUTES;
