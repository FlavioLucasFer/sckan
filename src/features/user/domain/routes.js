'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/users`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	SIGN_IN: `${BASE_URL}/sign-in`,
	LOGOUT: `${BASE_URL}/logout`,
};

export default ROUTES;
