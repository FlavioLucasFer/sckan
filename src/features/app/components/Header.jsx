import './css/Header.css';
import React from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import Icon from 'core/components/Icon';
import { HOME_ROUTE } from 'core/utils/routes';

import { setCurrentScreen } from 'redux/reducers/currentScreen';

function Header() {
	const { push } = useHistory();
	const dispatch = useDispatch();

	function handleLogoClick() {
		push(HOME_ROUTE);
		dispatch(setCurrentScreen(HOME_ROUTE));
	}

	return (
		<nav data-testid="navbar" className="white" style={{ height: '50px' }}>
			<div className="nav-wrapper">
				<a data-testid="app-logo" className="brand-logo center grey-text text-darken-3"
					onClick={handleLogoClick}>
					Sckan
				</a>
				<ul className="right hide-on-med-and-down nav-ul">
					<li data-testid="user-account" style={{ height: '50px' }}>
						<Icon type="round" 
							size="small"
							color="grey-text text-darken-3"
							style={{ height: '50px' }}>
							account_circle
						</Icon>
					</li>

					<li data-testid="more-options" style={{ height: '50px' }}>
						<Icon type="round"
							size="small"
							color="grey-text text-darken-3"
							style={{ height: '50px' }}>
							more_vert
						</Icon>
					</li>
				</ul>
			</div>
		</nav>
	);
}

export default Header;
