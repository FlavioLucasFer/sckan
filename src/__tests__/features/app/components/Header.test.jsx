import React from 'react';
import { shallow } from 'enzyme';

import Header from 'features/app/components/Header';
import { HOME_ROUTE } from 'core/utils/routes';

const mockHistory = {
	push: jest.fn(),
};

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom')),
	useHistory: () => (mockHistory),
}));

describe('Testing Header app feature component', () => {
	afterAll(() => {
		jest.resetAllMocks();
	});
	
	it('Should render correctly', () => {
		const wrapper = shallow(
			<Header />
		);

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('Should navigate to home screen when click on app logo', () => {
		const wrapper = shallow(
			<Header />
		);

		wrapper.find('.brand-logo').simulate('click');

		expect(mockHistory.push).toBeCalledWith(HOME_ROUTE);
	});
});