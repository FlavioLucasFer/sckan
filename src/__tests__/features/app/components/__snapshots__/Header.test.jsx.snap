// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header app feature component Should render correctly 1`] = `
"<nav data-testid=\\"navbar\\" className=\\"white\\" style={{...}}>
  <div className=\\"nav-wrapper\\">
    <a data-testid=\\"app-logo\\" className=\\"brand-logo center grey-text text-darken-3\\" onClick={[Function: handleLogoClick]}>
      Sckan
    </a>
    <ul className=\\"right hide-on-med-and-down nav-ul\\">
      <li data-testid=\\"user-account\\">
        <Icon type=\\"round\\" size=\\"small\\" color=\\"grey-text text-darken-3\\">
          account_circle
        </Icon>
      </li>
      <li data-testid=\\"more-options\\">
        <Icon type=\\"round\\" size=\\"small\\" color=\\"grey-text text-darken-3\\">
          more_vert
        </Icon>
      </li>
    </ul>
  </div>
</nav>"
`;
